applications/asset_tracker_v2/src/modules/modem_module.c

```diff
@@ -293,32 +293,22 @@ static int static_modem_data_get(void)
 
 	check_modem_fw_version();
 
-	struct modem_module_event *modem_module_event =
-			new_modem_module_event();
-
-	modem_module_event->data.modem_static.app_version =
-			CONFIG_ASSET_TRACKER_V2_APP_VERSION;
-
-	modem_module_event->data.modem_static.board_version =
-			modem_param.device.board;
-
-	modem_module_event->data.modem_static.modem_fw =
-			modem_param.device.modem_fw.value_string;
+	struct modem_module_event *modem_module_event = new_modem_module_event();
 
-	modem_module_event->data.modem_static.iccid =
-			modem_param.sim.iccid.value_string;
+	modem_module_event->data.modem_static.nw_mode_ltem = modem_param.network.lte_mode.value;
+	modem_module_event->data.modem_static.nw_mode_nbiot = modem_param.network.nbiot_mode.value;
+	modem_module_event->data.modem_static.nw_mode_gps = modem_param.network.gps_mode.value;
+	modem_module_event->data.modem_static.band = modem_param.network.current_band.value;
 
-	modem_module_event->data.modem_static.nw_mode_ltem =
-			modem_param.network.lte_mode.value;
+	strcpy(modem_module_event->data.modem_static.app_version,
```

Comment:
I think it would be safer to opt for `strncpy` for these string copies. It can happen that the destination buffers overflow, as we don't have full control over some of these string lengths

Reply:
Yes. Makes sense. Ill change to ```strncpy```.

Reply:
An option is also to validate that strings fit into buffers at compile time. Better to fail at build time rather than runtime whenever possible

Reply:
I assume we are talking a combination of ```BUILD_ASSERT``` and ```sizeof``` here?

Reply:
We could use build asserts when data from the ```modem_module``` module is put into the buffers in the ```data_module```. There it would make sense to use build asserts because string copies are done on char arrays for all cases. ````sizeof``` wouldn't work for ``` modem_param.device.board``` in this case because it is a char pointer. So we would need to mix and not be consistent for all the types we are testing for. We could use ```strncpy``` here and potentially migrate to build asserts later on if the format changes.

Reply:
Yeah, build asserts are not that important at this point. Most important is to ensure not to copy anything outside of buffer boundaries